### ISSUE TITLE
Turning on composefs breaks pivot from old bootimages

### ISSUE THREAD

**Issue by Prashanth684** [openshift/os#1678]
Attached the full secondboot journal. 
[journal.txt](https://github.com/user-attachments/files/18067503/journal.txt)

The specific error is:
```
Dec 09 17:13:23 localhost systemd[1]: Mounted /sysroot.
Dec 09 17:13:23 localhost systemd[1]: Starting OSTree Prepare OS/...
Dec 09 17:13:23 localhost ostree-prepare-root[675]: Resolved OSTree target to: /sysroot/ostree/deploy/fedora-coreos/deploy/8bd789c8d973d9910e8cc1fc477dd20ac5e975e0663085eeb06a95bfcdba5199.0
Dec 09 17:13:23 localhost ostree-prepare-root[675]: sysroot.readonly configuration value: 1 (fs writable: 1)
Dec 09 17:13:23 localhost ostree-prepare-root[675]: ostree-prepare-root: composefs: failed to mount: No such file or directory
Dec 09 17:13:23 localhost systemd[1]: ostree-prepare-root.service: Main process exited, code=exited, status=1/FAILURE
Dec 09 17:13:23 localhost systemd[1]: ostree-prepare-root.service: Failed with result 'exit-code'.
Dec 09 17:13:23 localhost systemd[1]: Failed to start OSTree Prepare OS/.
```

**Comment by jlebon** [comment]
This doesn't look like the first boot of the machine. Which AMI is this BTW? Is this from the CoreOS pipeline, or somewhere else? Let's try to reproduce it outside of OpenShift (presumably by booting the AMI, doing an `rpm-ostree rebase` or `bootc switch` and rebooting?).

**Comment by Prashanth684** [comment]
Yes this is not the first boot. This log is from the e2e test that is run on release nightlies, where the installer boots the machine with [fcos](https://github.com/openshift/installer/blob/master/data/data/coreos/fcos.json) AMis and then on second boot, pivots to scos.

**Comment by Prashanth684** [comment]
When i tried booting with the AMI that the Openshift installer uses - `39.20231101.3.0`,  I see this issue. But when booting with a newer AMI - `40.20240602.3.0`, I don't see the issue anymore and the rebase works. https://github.com/openshift/installer/pull/8640 updates the AMI for fcos and also tests the okd scos e2e.

**Comment by jlebon** [comment]
So the issue here is that any bootimage with a libostree older than v2023.4 will be too old to know how to create composefs images when upgrading. And so once we upgrade, `ostree-prepare-root` in the new initramfs fails because it sees that composefs should be enabled and yet no composefs image exists.

This should be fixed by https://github.com/ostreedev/ostree/pull/3353, which would allow ratcheting in composefs by letting `ostree-prepare-root` tolerate missing composefs images.

But clearly, longer-term we want to be able to rely on composefs just working the first time. This would definitely be fixed as we keep rolling out the bootimages update effort (https://github.com/openshift/enhancements/pull/1496), but we may decide to do a shorter-term tactical fix for it if we're blocked on it.

**Comment by jlebon** [comment]
For now, let's back out composefs: https://github.com/openshift/os/pull/1700

**Comment by dustymabe** [comment]
composefs was changed back to `maybe` in https://github.com/openshift/os/pull/1735

so it's enabled, but the `maybe` allows older bootimages.

**Comment by dustymabe** [comment]
longer term fix for this is:

- https://github.com/openshift/enhancements/pull/1761
